src/render/mod.rs

```diff
@@ -378,6 +425,21 @@ impl PipelineStates {
             skybox: pso_skybox,
         })
     }
+
+    pub(crate) fn pso_by_material(
+        &self,
+        material: &MaterialType,
+    ) -> &BasicPipelineState {
+        match *material {
+            MaterialType::Basic(_) => &self.mesh_basic_fill,
+            MaterialType::Line(_) => &self.line_basic,
+            MaterialType::Wireframe(_) => &self.mesh_basic_wireframe,
+            MaterialType::Lambert(_) => &self.mesh_gouraud,
+            MaterialType::Phong(_) => &self.mesh_phong,
+            MaterialType::Sprite(_) => &self.sprite,
+            _ => unreachable!(),
```

Comment:
Is it really unreachable or just an internal error if we hit this branch?

Reply:
Unreachable for sure (if somebody won't break the renderer logic).